railties/test/commands/routes_test.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "env_helpers"
+require "rails/command"
+require "rails/commands/routes/routes_command"
+
+class Rails::Command::RoutesTest < ActiveSupport::TestCase
+  setup :build_app
+  teardown :teardown_app
+
+  test "test_singular_resource_output_in_rake_routes" do
```

Comment:
If you use `test` method, you do not need to separate them with `_`. Therefore, you can change this like `singular resource output `.